### DISCUSSION TITLE
Adding new or stored Job to trigger

### DISCUSSION THREAD

**Discussion by CrazyMoka** [quartznet/quartznet#1660] [Q&A] [answered]
Hello,
Confused how to add a job to a trigger when its new or exists in the DB.

My code goes like this.

            var jobKey = new JobKey(_m.Name, "test1");

            var hasJob = scheduler.CheckExists(jobKey).Result;

            if (!hasJob)
            {

                job = JobBuilder.Create(jobClass)
                    .WithIdentity(jobKey)
                    .StoreDurably(true).Build();
            }

            TriggerBuilder trigger = TriggerBuilder.Create();
            trigger.ForJob(jobKey);
            trigger.WithIdentity("instant", "test1");

            trigger.StartNow();
            ITrigger _trigger = trigger.Build();

            await scheduler.ScheduleJob(_trigger); /// issue line. If the job is new this fails.

What is the proper way to add a job to a trigger if its durable or new?
Assume the trigger isn't running.

Thanks

**Comment by CrazyMoka** [comment, marked as answer]
in the If has Job ststement I added schedule.AddJob(_job,true);
That seems to have worked. But is that correct?

**Reply by lahma** [reply]
`AddJob` should do the trick if you just want to make sure that job exists in DB and stays there regardless of the triggers (having the durable set for job). Please note that accessing `.Result` is a bad practice and can cause deadlocks, you should always `await` the tasks.

**Reply by CrazyMoka** [reply]
Hi, thank you for the response.  When I used await it would give me an error further down in my code about object not being set. So I used .Result and it fixed the issue.

**Reply by CrazyMoka** [reply]
Never mind await seems to be working now... coding is fun.
